org.eclipse.winery.model.threatmodeling/src/main/java/org/eclipse/winery/model/threatmodeling/ThreatModelingUtils.java

```diff
@@ -0,0 +1,353 @@
+            if (Objects.nonNull(serviceTemplatePolicies)) {
+                mitigationPolicies = serviceTemplatePolicies.getPolicy()
+                    .stream() // check if polcies are mitigation policies
+                    .filter((TPolicy policy) -> policy.getPolicyType().toString().equals(ThreatModelingConstants.MITIGATION_POLICY_ID))
+                    .collect(Collectors.toList());
+            }
+        }
+
+        return mitigationPolicies;
+    }
+
+    /**
+     * Retreive all threat Policy Templates available in the current repository
+     *
+     * @return a list of all threat policy templates
+     */
+    public List<Threat> getThreatCatalogue() {
+
+        return this.repository.getAllDefinitionsChildIds(PolicyTemplateId.class)
+            .stream()
+            .map(repository::getElement)
+            .filter((TPolicyTemplate policyTemplate) -> policyTemplate.getTypeAsQName().toString().equals(ThreatModelingConstants.THREAT_POLICY_ID))
+            .map((TPolicyTemplate policyTemplate) -> {
+                Threat threat = new Threat();
+                if (Objects.nonNull(policyTemplate.getProperties())) {
+                    threat.setProperties(policyTemplate.getProperties().getKVProperties());
+                }
+                // LOGGER.info("{}", policyTemplate.getName());
+                threat.setTemplateName(policyTemplate.getName());
+                threat.setNamespace(policyTemplate.getTypeAsQName().getNamespaceURI());
+                return threat;
+            })
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Creates matching threat mitigations policy template pairs.
+     *
+     * @return status string passed to user
+     * @throws IOException when persistence fails
+     */
+    public String createThreatAndMitigationTemplates(String name, String description, String stride, String severity) throws IOException {
+        
+        /* check prerequisites for implicit setup */
+        if (!checkPrerequisites()) {
+            try {
+                setupThreatModelingTypes();
+            } catch (Exception e) {
+                return "Error while setting up threat modeling types";
+            }
+        }
+        
+        /* Create Threat */
+        TPolicyTemplate threat = new TPolicyTemplate();
+        QName threatTypeQName = QName.valueOf(ThreatModelingConstants.THREAT_POLICY_ID);
+
+        String threatName = Util.makeNCName(name);
+        PolicyTemplateId threatID = BackendUtils.getDefinitionsChildId(PolicyTemplateId.class, ThreatModelingConstants.THREATMODELING_NAMESPACE, threatName, false);
+        threat.setId(threatName);
+        threat.setName(threatName);
+        threat.setType(threatTypeQName);
+
+        TPolicyTemplate.Properties threatProps = new TPolicyTemplate.Properties();
+        Map<String, String> propMap = new HashMap<>();
+        propMap.put("description", description);
```

Comment:
You could use an enum or variables for these keys

Reply:
See comment above